devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -0,0 +1,572 @@
+/*
+ *  Copyright 2017 - 2018 Stelpro
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Stelpro Ki Thermostat
+ *
+ *  Author: Stelpro
+ *
+ *  Date: 2018-04-24
+ */
+
+metadata {
+	definition (name: "Stelpro Ki Thermostat", namespace: "stelpro", author: "Stelpro", ocfDeviceType: "oic.d.thermostat") {
+		capability "Actuator"
+		capability "Temperature Measurement"
+		capability "Temperature Alarm"
+		capability "Thermostat"
+		capability "Thermostat Mode"
+		capability "Thermostat Operating State"
+		capability "Thermostat Heating Setpoint"
+		capability "Configuration"
+		capability "Polling"
+		capability "Sensor"
+		capability "Refresh"
+		capability "Health Check"
+
+		attribute "outsideTemp", "number"
+
+		command "switchMode"
+		command "quickSetHeat"
+		command "quickSetOutTemp"
+		command "increaseHeatSetpoint"
+		command "decreaseHeatSetpoint"
+		command "setCustomThermostatMode"
+		command "eco"
+		command "updateWeather"
+
+		fingerprint deviceId: "0x0806", inClusters: "0x5E,0x86,0x72,0x40,0x43,0x31,0x85,0x59,0x5A,0x73,0x20,0x42", mfr: "0239", prod: "0001", model: "0001", deviceJoinName: "Stelpro Ki Thermostat"
+	}
+
+	// simulator metadata
+	simulator { }
+
+	preferences {
+		input("heatdetails", "enum", title: "Do you want a detailed operating state notification?", options: ["No", "Yes"], defaultValue: "No", required: true, displayDuringSetup: true)
+		input("zipcode", "text", title: "ZipCode (Outdoor Temperature)", description: "[Do not use space](Blank = No Forecast)")
+	}
+
+	tiles(scale : 2) {
+		multiAttributeTile(name:"thermostatMulti", type:"thermostat", width:6, height:4, canChangeIcon: true) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState("temperature", label:'${currentValue}°')
+			}
+			tileAttribute("device.heatingSetpoint", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "increaseHeatSetpoint")
+				attributeState("VALUE_DOWN", action: "decreaseHeatSetpoint")
+			}
+			tileAttribute("device.thermostatOperatingState", key: "OPERATING_STATE") {
+				attributeState("idle", backgroundColor:"#44b621")
+				attributeState("heating", backgroundColor:"#ffa81e")
+			}
+			tileAttribute("device.thermostatMode", key: "THERMOSTAT_MODE") {
+				attributeState("heat", label:'${name}')
+				attributeState("eco", label:'${name}')
```

Comment:
I have misgivings about this working in OneApp

Reply:
It does!